src/plan/global.rs

```diff
@@ -601,33 +601,33 @@ impl<VM: VMBinding> BasePlan<VM> {
         pages
     }
 
-    pub fn trace_object<T: TransitiveClosure>(
+    pub fn trace_object<Q: ObjectQueue>(
```

Comment:
Not sure if it is helpful to mark this as `#[cold]` or `#[inline(never)]`. This may reduce the inlined code size of the hot trace_object calls.

Reply:
Yes.  It may reduce the inline code size.  But I am not sure if we should mark it as `#[cold]` or `#[inline(never)]`.

1.  The VM space (if exists) may not be cold.
2.  If `in_space` is cheap and simple, inlining it should help skipping objects that are not in those spaces.